### DISCUSSION TITLE
Tasmota 14.3.0 on ESP8266: RULE ON Shutter1#Position DO VAR1 %value% ENDON flooding console (even after shutter movement)?

### DISCUSSION THREAD

**Discussion by forumlw06** [arendst/Tasmota#22338] [Q&A] [answered]
Hello,

was there a change in 14.3.0 in rule processing for shutters?

I have this rule active to save shutterposition in a variable: 
`rule3 ON Shutter1#Position DO VAR1 %value% ENDON`

Basically this works fine and i get the shutterposition in VAR1.

However until Tasmota 14.2.0 rule3 did no longer trigger after the shutter was moved to the desired position.

Looks like in 14.3.0 the rule seems to trigger continuosly after the desired position is reached. Not a real problem (except that variable  is set and set again), but this also floods the console .... See following console output

18:51:00.117 CMD: shutterposition 10
18:51:00.125 RSL: RESULT = {"POWER1":"ON"}
18:51:00.128 RSL: POWER1 = ON
18:51:00.131 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":10,"Tilt":0}}
18:51:00.176 RUL: SHUTTER1#POSITION performs 'VAR1 0'
18:51:00.182 RSL: RESULT = {"Var1":"0"}
18:51:00.296 RUL: POWER1#STATE=1 performs 'VAR2 1'
18:51:00.301 RSL: RESULT = {"Var2":"1"}
18:51:00.347 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":10,"Tilt":0}}
18:51:00.392 RUL: SHUTTER1#POSITION performs 'VAR1 0'
**...... (i removed some intermediate values to shorten log)**
18:51:07.228 RSL: RESULT = {"Shutter1":{"Position":4,"Direction":1,"Target":10,"Tilt":0}}
18:51:07.272 RUL: SHUTTER1#POSITION performs 'VAR1 4'
**...... (i removed some intermediate values to shorten log)**
18:51:10.238 RSL: RESULT = {"Shutter1":{"Position":8,"Direction":1,"Target":10,"Tilt":0}}
18:51:10.283 RUL: SHUTTER1#POSITION performs 'VAR1 8'
18:51:10.288 RSL: RESULT = {"Var1":"8"}
18:51:10.821 RSL: RESULT = {"POWER1":"OFF"}
18:51:10.824 RSL: POWER1 = OFF
18:51:10.827 RSL: SHUTTER1 = 11
18:51:10.830 RSL: RESULT = {"Shutter1":{"Position":11,"Direction":0,"Target":10,"Tilt":0}}
18:51:10.874 RUL: SHUTTER1#POSITION performs 'VAR1 11'
18:51:10.880 RSL: RESULT = {"Var1":"11"}
**i would have expected that the rule will not trigger beyond this point, but it continues without stopping**
18:51:10.928 RUL: SHUTTER1#POSITION performs 'VAR1 11'
18:51:10.934 RSL: RESULT = {"Var1":"11"}
18:51:11.771 RUL: SHUTTER1#POSITION performs 'VAR1 11'
18:51:11.776 RSL: RESULT = {"Var1":"11"}
18:51:12.708 RUL: SHUTTER1#POSITION performs 'VAR1 11'
18:51:12.714 RSL: RESULT = {"Var1":"11"}

How can is stop the rule from triggering after desired shutterposition was reached?

Thanks for any hint!

Rainer

**Comment by sfromis** [comment, marked as answer]
`rule3 ON Shutter1#Position!=%var1% DO VAR1 %value% ENDON` will only fire when the value changes.

**Reply by forumlw06** [reply]
Thanks for the quick response! 

I updated the ruleset and will test it.

Just out of curiosity: something must have been changed in rule processing between Tasmota <= 14.2.0 and 14.3.0.
The original rule was running without that permanent trigger before the update. I checked the changelog of 14.3.0, but wasn't able to identify a possible change that would cause this

**Reply by sfromis** [reply]
@stefanbode is maintaining the shutters driver, with quite some change activity, maybe something worked out a bit differently.

**Reply by sfromis** [reply]
BTW. See https://github.com/arendst/Tasmota/issues/22282

**Reply by forumlw06** [reply]
Thanks for the additional information!
